genesyscloud/telephony_providers_edges_site/genesyscloud_telephony_providers_edges_site_proxy.go

```diff
@@ -0,0 +1,454 @@
+func (p *siteProxy) createSiteOutboundRoute(ctx context.Context, siteId string, outboundRoute *platformclientv2.Outboundroutebase) (*platformclientv2.Outboundroutebase, error) {
+	return p.createSiteOutboundRouteAttr(ctx, p, siteId, outboundRoute)
+}
+
+// getSiteByIdFunc returns a single Outbound Route by Id
+func (p *siteProxy) getSiteOutboundRoutes(ctx context.Context, siteId string) (*[]platformclientv2.Outboundroutebase, error) {
+	return p.getSiteOutboundRoutesAttr(ctx, p, siteId)
+}
+
+// updateSiteFunc updates a Genesys Cloud Outbound Route for a Genesys Cloud Site
+func (p *siteProxy) updateSiteOutboundRoute(ctx context.Context, siteId string, outboundRouteId string, outboundRoute *platformclientv2.Outboundroutebase) (*platformclientv2.Outboundroutebase, error) {
+	return p.updateSiteOutboundRouteAttr(ctx, p, siteId, outboundRouteId, outboundRoute)
+}
+
+// deleteSiteFunc deletes a Genesys Cloud Outbound Route by Id for a Genesys Cloud Site
+func (p *siteProxy) deleteSiteOutboundRoute(ctx context.Context, siteId string, outboundRouteId string) (*platformclientv2.APIResponse, error) {
+	return p.deleteSiteOutboundRouteAttr(ctx, p, siteId, outboundRouteId)
+}
+
+// getSiteNumberPlansFunc retrieves all Number Plans of a Genesys Cloud Sites
+func (p *siteProxy) getSiteNumberPlans(ctx context.Context, siteId string) (*[]platformclientv2.Numberplan, *platformclientv2.APIResponse, error) {
+	return p.getSiteNumberPlansAttr(ctx, p, siteId)
+}
+
+// updateSiteNumberPlansFunc updates the Number Plans for a Genesys Cloud Site
+func (p *siteProxy) updateSiteNumberPlans(ctx context.Context, siteId string, numberPlans *[]platformclientv2.Numberplan) (*[]platformclientv2.Numberplan, *platformclientv2.APIResponse, error) {
+	return p.updateSiteNumberPlansAttr(ctx, p, siteId, numberPlans)
+}
+
+// getLocation retrieves a Genesys Cloud Location by Id
+func (p *siteProxy) getLocation(ctx context.Context, locationId string) (*platformclientv2.Locationdefinition, error) {
+	return p.getLocationAttr(ctx, p, locationId)
+}
+
+// getTelephonyMediaregions retrieves the Genesys Cloud media regions
+func (p *siteProxy) getTelephonyMediaregions(ctx context.Context) (*platformclientv2.Mediaregions, error) {
+	return p.getTelephonyMediaregionsAttr(ctx, p)
+}
+
+// setDefaultSite sets a Genesys Cloud Site as the default site for the org
+func (p *siteProxy) setDefaultSite(ctx context.Context, siteId string) error {
+	return p.setDefaultSiteAttr(ctx, p, siteId)
+}
+
+// getDefaultSiteId gets the default Site for the Genesys Cloud org
+func (p *siteProxy) getDefaultSiteId(ctx context.Context) (siteId string, err error) {
+	return p.getDefaultSiteIdAttr(ctx, p)
+}
+
+// getAllManagedSitesFn is an implementation function for retrieving all Genesys Cloud Outbound managed Sites
+func getAllManagedSitesFn(ctx context.Context, p *siteProxy) (*[]platformclientv2.Site, error) {
+	var allManagedSites []platformclientv2.Site
+
+	for pageNum := 1; ; pageNum++ {
```

Comment:
We need to check the number of pages first to avoid complications in the future. Look the the `getAllPhonesFn` function in`genesyscloud_telephony_providers_edges_phone_proxy.go` to see what I mean on this

Reply:
fixed